### DISCUSSION TITLE
Use `router.subscribe` in order to monitor route changes

### DISCUSSION THREAD

**Discussion by Miz85** [remix-run/react-router#11216] [Q&A] [answered]
Hi!

I'd like to create a library to monitor and track routes changes. I noticed that the `Router` type has a `subscribe` method that allows you to subscribe to router state changes but it's marked as `PRIVATE - do not use`. Is it an absolute no-no or would it be okay in the context of creating tooling that integrates with React Router?

**Comment by Miz85** [comment, marked as answer]
I'm gonna close this question as I got [an answer](https://discord.com/channels/770287896669978684/1006299623083741185/threads/1166772402055815228) on the Remix discord

**Reply by jacobjuul** [reply]
would be nice to share here. That link is already dead

**Reply by sergiodxa** [reply]
The links work for me, but here's a copy:

> `router.subscribe` is largely safe to use.  Those are mostly marked private to avoid folks reaching for them inside their react router app - but that's a valid use-case to listen for changes from outside the react tree.  We labelled them as such more as a precaution when we released that `@remix-run/router` package.
> However, another option would be to use an effect on the `useLocation` value and trigger a custom DOM event - that way you avoid the `router` entirely.
